OneCommunityGlobal/HGNRest
# Bailey new personal max backend

# Description
18. (PRIORITY High) Bailey/Jae: Fix the New Max Personal Record award.
The badge should just update the number next to it AND on it, and not be awarded again in the list shown below.
Every time the record is broken though, it should still show up in the New Badges section of the person’s Dashboard


## Related PRS (if any):
None

## Main changes explained:
- Added badgeCollection of only Personal Max badges to badgeOfType since badgeOfType was defined but was assigned no value.
- Created an empty array called duplicateBadges for the duplicateBadges to be pushed into and removed without removing all Personal Max Badges.
- Improved the removeDupBadge functionality to look for the objectId of the badge rather than just the badgeId.


## How to test:
1. check into current branch
2. do `npm install`  to run this PR locally
3. Navigate to cronjobs folder→userProfileJobs→change the cronjob from '1 0 * * * ' to '* * * * *' so it runs every minute.
4. Open MongoDb compass and find your user profile that you want to test and change personalBestMaxHrs to a value that is greater than 10→change lastWeekTangibleHrs to match the personalBestMaxHrs.
5. do `npm run build && npm start` This ensures the changes for the cronjob go into affect so it will run right away.
6. Login as your test account and Empty cache and hard reload so the changes appear on the page under Badges for the front end.

## Screenshots or videos of changes:
Walkthrough of how to test the back end.
[Personal Max Testing Guide.webm](https://github.com/OneCommunityGlobal/HGNRest/assets/108101478/09629d0b-9288-429d-b842-fa6fe4d93fb0)

The fields to change in the MongoDB database
![Showing what to change in MongoDB](https://github.com/OneCommunityGlobal/HGNRest/assets/108101478/e53b74b4-862b-4964-842f-236fafce1e64)


## Note:
The badge only awards on Sundays, this testing method just helps award the badge right away to make sure the badge does not duplicate. 

When awarding the badge right away it will not appear under new badges until Sunday and the count of the badges inside the report badges will not update until Sunday as well.
